### ISSUE TITLE
Osmand should follow bike roads

### ISSUE THREAD

**Issue by Rhgitx** [osmandapp/OsmAnd#16952]
I  would like to have a navigation/route planning feature which  would allow osmand to prefer official bike routes. 
Please  have a look at  Android App bikemap, which does have such a feature.
It is annoying when you see the bike routes displayed but can not use them for route planning!

**Comment by albansuser** [comment]
Hello

That's not my experience.

The Cycle routing function (as opposed to Car, Walk etc)  has four Riding Style options. One of these is likely to meet your requirements. Have you tried them?
 
I find that "Prefer byways" generally meets my requirements for giving priority to cycle paths. Alternatively I  use the excellent BRouter plugin.

Good luck

**Comment by Zirochkabila** [comment]
@Rhgitx Try to make it this way: Navigation > Set start & finish > Settings > Riding style

**Comment by Rhgitx** [comment]
Hello,
thanks for mentioning Brouter, I was not aware of it. Yes, I tried the 4 cycling options but did not meet my expectations.
If I deselect "fastest route" and using Brouter this seems to do the optimal routing..
I will try it on my next bike tours..

12.04.2023 06:11:41 albansuser ***@***.***>:

**Comment by albansuser** [comment]
Yes, in my view, the BRouter plugin completes the set of useful Riding Style options.

Surely an internal equivalent of BRouter giving highest priority to using designated cycle paths could be implemented.

**Comment by Iey4iej3** [comment]
None of these four modes seem to have a preference for bike lanes. I find it much safer to cycle on bike lanes, where I do not compete with cars and I feel much safer.

**Comment by albansuser** [comment]
In my experience the "prefer byways" bike routing setting gives a reasonable level of preference to cycle paths.

However the BRouter plugin results in a high level of preference for cycle paths and therefore is often my choice of route.

In the attached examples, the Brouter route (55 km) is almost totally on cycle paths whereas the other is a compromise mix of  secondary roads and cycle paths.

![Screenshot_20240719_110734_OsmAnd~](https://github.com/user-attachments/assets/f39ac88f-2164-479f-8080-ee66198b83d9)
![Screenshot_20240719_110820_OsmAnd~](https://github.com/user-attachments/assets/b0682f57-1030-4b86-92f9-f065a6292a84)
